blog/2023-10-10-weaviate-1-22-release/index.mdx

```diff
@@ -0,0 +1,253 @@
+
+### Future improvements
+
+Please keep in mind that this is the first iteration of nested JSON object storage in Weaviate, and we have further improvements planned.
+
+In the initial implementation, any properties with `object` datatype do not affect the object vector, or the inverted index. This behavior is similar to what would happen with a cross-reference property, where the cross-referenced object is not indexed.
+
+Future plans include the ability to index nested properties, for example to allow for filtering on nested properties and vectorization options.
+
+## Async vector indexing
+
+Weaviate now offers the option of asynchronous vector indexing to object creation. In turn, it enables you to optimize imports for speed where vector indexing is not required immediately.
+
+In the past, the HNSW index would be built synchronously with object creation. This meant that the import process would occur in lockstep with the indexing process. The benefit of the synchronous approach is that the user can perform a search immediately after import, as the data is already indexed. However, if your only goal was fastest data ingestion, this could slow down the import process.
+
+With `1.22`, the vector indexing process can now be performed asynchronously to object creation. This means that the import process can go on independently to the indexing process and continue to create objects while the indexing process is ongoing.
+
+Outwardly, this change does not affect the user experience, other than the import process returning a response faster than before.
+
+The user can still perform a search immediately after import, as Weaviate will search any un-indexed data using brute force. This means that the user will not have to wait for the indexing process to complete before being able to search the newly imported data.
+
+The `nodes` REST endpoint has been updated to reflect this change.
+
+`INDEXING` has been added to the list of available node statuses, and the `stats` field now includes an `vectorQueueLength` field, which contains the number of objects currently being indexed.
+
+As of `1.22`, this is an opt-in feature. It can be enabled by setting the `WEAVIATE_ASYNC_INDEXING` setting to `true`, such as in the `docker-compose.yml` file.
+
+## Additional gRPC support
+
+The team continues to add gRPC support under the hood, with a big part of the equivalent GraphQL `Get` queries, and batch import functionality now available via gRPC.
+
+With these changes, we have seen improvements in import speeds of around ~60-80% for batch imports, and multiple times faster `Get` queries. A big part of this comes from the fact that gRPC is a binary protocol, which is more efficient than the text-based GraphQL protocol, as well as it being based on HTTP/2.
```

Comment:
Consider:
imports. `Get` queries are many times faster, too.


And
A big part of this speedup is because gRPC is a binary, HTTP/2 protocol that is more efficient than the text-based GraphQL protocol.